common/src/main/java/com/wynntils/services/map/Label.java

```diff
@@ -41,29 +47,44 @@ public LabelLayer getLayer() {
         return LabelLayer.values()[layer - 1];
     }
 
-    /**
-     * The recommended minimum combat level for visiting this place
-     */
-    public String getLevel() {
-        if (level == null) return "";
-
+    public int getLevel() {
         return level;
     }
 
-    public int getCombatLevel() {
-        if (level == null) return 0;
-
-        try {
-            return Integer.parseInt(level);
-        } catch (NumberFormatException e) {
-            return 0;
-        }
-    }
-
     public Location getLocation() {
         return new Location(x, 0, z);
     }
 
+    public static final class LabelDeserializer implements JsonDeserializer<Label> {
```

Comment:
When we migrate to the new mapdata providers, I think we can just enforce having label as an integer instead of a string. I don't recall the details, but apparently thenew system reuses the services.map.Label and PoiService..? Maybe we should split it off so we can have a fresh start with an integer instead of a string for the json.

Reply:
I've been thinking about dropping the Label class. I've actually had to figure out how a "label" turns into a map feature. I thought we just implement the map
feature interface on the label class, but no. It is wrapped in a "service provider". I wonder that if the service class should not use **this** label class, but just a temporary deserialization class.

Reply:
Hm, I'm vaguely recalling this logic. I think the Label class was created to match the Wynncraft web map data with the place labels. When we switch to proper MapData, this will be redone as a normal json file. I think that was one of my actual test cases when designing the MapData structure/json format, that this should be possible.

Reply:
Okay, I think that is good. That is, for the Label part. We still need to figure out the level-0 thing.